netty/src/main/java/io/grpc/netty/ProtocolNegotiators.java

```diff
@@ -128,22 +132,44 @@ public AsciiString scheme() {
     };
   }
 
+  /**
+   * Create a server TLS handler for HTTP/2 capable of using ALPN/NPN.
+   */
+  public static ProtocolNegotiator serverTls(final SslContext sslContext) {
+    return serverTls(sslContext, null);
+  }
+
   static final class ServerTlsHandler extends ChannelInboundHandlerAdapter {
     private final ChannelHandler next;
     private final SslContext sslContext;
+    private Executor executor;
+    private ObjectPool<? extends Executor> executorPool;
 
     private ProtocolNegotiationEvent pne = ProtocolNegotiationEvent.DEFAULT;
 
-    ServerTlsHandler(ChannelHandler next, SslContext sslContext) {
+    ServerTlsHandler(ChannelHandler next,
+        SslContext sslContext,
+        ObjectPool<? extends Executor> executorPool) {
       this.sslContext = checkNotNull(sslContext, "sslContext");
       this.next = checkNotNull(next, "next");
+      this.executorPool = executorPool;
+    }
+
+    ServerTlsHandler(ChannelHandler next, SslContext sslContext) {
+      this(next, sslContext, null);
     }
 
     @Override
     public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
       super.handlerAdded(ctx);
       SSLEngine sslEngine = sslContext.newEngine(ctx.alloc());
-      ctx.pipeline().addBefore(ctx.name(), null, new SslHandler(sslEngine, false));
+      if (this.executorPool != null) {
```

Comment:
You removed the check for `"io.netty.handler.ssl.openssl.useTasks"` system property. May be it is okay - but just curious why it was there and why you removed it now.

Reply:
According to this PR, https://github.com/netty/netty/pull/8847, we need to set this system property in order to make the extra Executor to take effect. It is used for offloading time-consuming TLS operations.
Not setting this property would leave the Executor unused, which falls back to the original implementation, though the overhead of getting executor might be a concern. 
My earlier intention is that, if users didn't specify this property, we'd better use the old API to avoid the overhead. But after a while I realized that checking this property at handler adding time has nothing to do with whether it is set when actually needed ... This system property is a global configuration and is subject to change at any time. Adding an additional check here doesn't help that much, IIUC.
My current plan is, we will finally have a user documentation, which will ask users to set this property if using any time-consuming TLS operations. @ejona86 please correct me if I misunderstand something or overthinking this because I remembered we also talked about this before :-)